src/blocks/sandbox-image/imports/index.mjs

```diff
@@ -0,0 +1,324 @@
+/**
+ * BU's take on the Image component.
+ */
+
+// External dependencies.
+import classnames from 'classnames';
+import { useState } from '@wordpress/element';
+
+import {
+	MediaPlaceholder,
+	InspectorControls,
+	MediaUploadCheck,
+	MediaUpload,
+} from '@wordpress/block-editor';
+
+import {
+	Button,
+	IconButton,
+	FocalPointPicker,
+	PanelBody,
+	PanelRow,
+	Placeholder,
+} from '@wordpress/components';
+
+import { __ } from '@wordpress/i18n';
+
+// BU dependencies.
+import {
+	useMedia,
+	getImageData,
+	LoadingSpinner,
+} from '@bostonuniversity/block-imports';
+
+/**
+ * Returns the class list for the component based on the current settings.
+ *
+ * @param {string} className Additional classes assigned to the component.
+ */
+const getClasses = ( className ) =>
+	classnames( 'bu-components-image', {
+		[ className ]: className,
+	} );
+
+/**
+ * Export component.
+ *
+ * @param  props
+ * @return {Element} Element to render, in this case an DIV.
+ */
+export const Image = ( props ) => {
+	const {
+		// Setup
+		canEditImage = true,
+		debug = true,
+		tag = 'img',
+		size = 'thumbnail',
+		sourceSizes = [],
+		altSource = 'alt',
+		mediaId = undefined,
+		className = undefined,
+		canEditFocalPoint = true,
+		focalPoint = { x: 0.5, y: 0.5 },
+		// Placeholder
+		label = 'Placeholder label...',
+		instructions = 'Placeholder instructions...',
+		// MediaPlaceholder
+		labels = {
+			title: 'MediaPlaceholder label...',
+			instructions: 'MediaPlaceholder instructions...',
+		},
+		//
+		...rest
+	} = props;
+
+	/**
+	 * States.
+	 */
+	const [ currentFocalPoint, setFocalPoint ] = useState( focalPoint );
+	const [ currentMediaId, setMediaId ] = useState( mediaId );
+
+	let imgObj = false;
+
+	// Fetch the media object based on the `currentMediaId`. Should this be useEffect?
+	const { mediaObj, isResolvingMedia, hasResolvedMedia } =
+		useMedia( currentMediaId ); // @todo so this returns a different object than the media picker; its a waste of resources to get the entire media object from the media picker and then call this function, but they have different structure... optimize somehow?
+
+	// If Debug is set to true, output some helpful information to the console for block developers to utilize media object info in their block development.
+	if ( isResolvingMedia ) {
+		if ( debug ) {
+			// eslint-disable-next-line no-console
+			console.log( 'Image Media Fetch in Progress: ', isResolvingMedia );
+		}
+		return (
+			<LoadingSpinner
```

Comment:
This LoadingSpinner is meant to be used overlaid over content... such as centered over something. 

You could try setting `shadow={false}` on it (or maybe we should just make that the default)

Alternatively we can just use the core Spinner or perhaps load this positioned on top of some placeholder element that represents the image that is being loaded? Like a grayed out box or something...